tensordict/tensordict.py

```diff
@@ -2529,18 +2529,43 @@ def contiguous(self) -> TensorDictBase:
             return self.clone()
         return self
 
-    def select(self, *keys: str, inplace: bool = False) -> TensorDictBase:
-        d = {key: value for (key, value) in self.items() if key in keys}
-        d_meta = {
-            key: value
-            for (key, value) in self.items_meta(make_unset=False)
-            if key in keys
-        }
+    def select(self, *keys: NESTED_KEY, inplace: bool = False) -> TensorDictBase:
+        for key in keys:
+            if key not in self.keys(include_nested=True):
```

Comment:
I would suggest to minimize the creation of the key object, as it's likely to be quite expensive. 
Why not
```python
        existing_keys = set(self.keys(include_nested=True))
        for key in keys:
            if key not in existing_keys:
                raise KeyError(f"Key {key} was not found.")
```

Reply:
I've added this without the `set` because otherwise we run into problems in the recursion with tuples of length 1 not being found in the set of keys (because in `_TensorDictKeysView.__iter__` they are automatically swapped out for strings.

I think this is fine because the `__contains__` method is not generating all keys each time, it's doing a series of O(1) lookups of keys so still efficient, and `__contains__` handles tuples of length 1 automatically for us.